### ISSUE TITLE
I'm having an timeout issue, specifically a WebException.

### ISSUE THREAD

**Issue by mhughey-vent** [xamarin/xamarin-macios#5992]
### Steps to Reproduce

So the issue I'm running into is a WebException (specifically a timeout).

I have been able to narrow it down to the HTTP Client implementation settings in iOS.

- When it is set to **Managed** I don't get the error

- When I set it to **NSUrlSession (iOS 7+)** a timeout exception is being generated after a request has been running longer than 1 minute.

We also tested our json request, using Insomnia, and it took about 5 mins to complete and we got data, but no exception is generated.

I do not have access to the middle tier, just the server logs that get updated.

I know about the HttpClient.Timeout property which we have set for large amount of time, are there any other configuration variables for setting the timeout when needing TLS 1.2?

(The TLS 1.2 is a requirement)

_:: Extra Info That May Be Helpful ::
It looks as though NSUrlSession (iOS 7+) setting forces the native HTTP Client to be used (which has a hard-coded timeout of 60 seconds)_

This was cross posted at Xamarin.Forms before I new where to post.

### Expected Behavior
A HTTP response, specifically a  file (size may vary)

### Actual Behavior
I get a WebException (Request timeout)

### Environment
Mac OS
- XCode 8.3.3

Visual Studio 2015 for Windows
- Xamarin 4.6.3.4
- Xamarin.IOS 10.12.3.3

```
Microsoft Visual Studio Professional 2015
Version 14.0.25420.01 Update 3
Microsoft .NET Framework
Version 4.7.03056

Installed Version: Professional

LightSwitch for Visual Studio 2015   00325-60003-96533-AA506
Microsoft LightSwitch for Visual Studio 2015

Visual Basic 2015   00325-60003-96533-AA506
Microsoft Visual Basic 2015

Visual C# 2015   00325-60003-96533-AA506
Microsoft Visual C# 2015

Visual C++ 2015   00325-60003-96533-AA506
Microsoft Visual C++ 2015

Visual F# 2015   00325-60003-96533-AA506
Microsoft Visual F# 2015

Windows Phone SDK 8.0 - ENU   00325-60003-96533-AA506
Windows Phone SDK 8.0 - ENU

Application Insights Tools for Visual Studio Package   7.0.20622.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2015.1 (Beta8)   14.1.11107.0
ASP.NET and Web Tools 2015.1 (Beta8)

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
For additional information, visit http://www.asp.net/

Clang with Microsoft CodeGen   14.0.25516
Clang with Microsoft CodeGen

Common Azure Tools   1.8
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GitHub.VisualStudio   2.9.0.7614
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

KofePackagePackage Extension   1.0
KofePackagePackage Visual Studio Extension Detailed Info

Merq   1.1.17-rc (cba4571)
Command Bus, Event Stream and Async Manager for Visual Studio extensions.

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Mono Debugging for Visual Studio   4.6.8-pre (ec7034f)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   3.4.4
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

Office Developer Tools for Visual Studio 2015 ENU   14.0.23025
Microsoft Office Developer Tools for Visual Studio 2015 ENU

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

Python Tools for Visual Studio   2.2.50113.00
Python Tools for Visual Studio provides IntelliSense, projects, templates, Interactive windows, and other support for Python developers.

Python Tools for Visual Studio - Django Integration   2.2.50113.00
Provides templates and integration for the Django web framework.

Python Tools for Visual Studio - Profiling Support   2.2.50113.00
Profiling support for Python projects.

SQL Server Data Tools   14.0.60519.0
Microsoft SQL Server Data Tools

TypeScript   1.8.34.0
TypeScript tools for Visual Studio

Visual C++ for Cross Platform Mobile Development (Android)   14.0.25401.00
Visual C++ for Cross Platform Mobile Development (Android)

Visual C++ for Cross Platform Mobile Development (iOS)   14.0.25401.00
Visual C++ for Cross Platform Mobile Development (iOS)

Visual Studio Tools for Apache Cordova   Update 10
Visual Studio Tools for Apache Cordova

Visual Studio Tools for Universal Windows Apps   14.0.25527.01
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.

Workflow Manager Tools 1.0   1.0
This package contains the necessary Visual Studio integration components for Workflow Manager.

Xamarin   4.6.3.4 (0841c2aaf)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin.Android   7.4.3.1 (18558b4)
Visual Studio extension to enable development for Xamarin.Android.

Xamarin.iOS   10.12.3.3 (8a39106)
Visual Studio extension to enable development for Xamarin.iOS.
```

**Comment by wesvf** [comment]
Is ModernHttpClient being used behind the scenes to enable support for TLS 1.2? If so there is hard limit on timeouts in iOS that has been rasied as an issue previously but not been addressed ([https://github.com/anaisbetts/ModernHttpClient/issues/261](url), [https://github.com/anaisbetts/ModernHttpClient/issues/21](url)).

We've had to fork our requests that we expect to take longer to get around this, but when applying the TLS setting as described here ([https://docs.microsoft.com/en-us/xamarin/cross-platform/app-fundamentals/transport-layer-security?tabs=windows](url)) it seems to bypass our application logic.

**Comment by mhughey-vent** [comment]
Here's a test project but I haven't found a online REST API to simulate a long request greater than 60 secs.

Here's a online REST API website, I thought I could use but it has a hard cut off of 30 secs - http://slowwly.robertomurray.co.uk/
[TestLongRunningRequest.SLN.zip](https://github.com/xamarin/xamarin-macios/files/3148891/TestLongRunningRequest.SLN.zip)

**Comment by spouliot** [comment]
This looks like a duplicate of https://github.com/xamarin/xamarin-macios/issues/5190 which we fixed last year. The fix is part of our 16.0 milestone (the current stable XI). However

> Xamarin.iOS   10.12.3.3 (8a39106)

is a very old version of Xamarin.iOS. You'll need to update to the latest stable release to get the fix.

**Comment by spouliot** [comment]
Assuming this is fixed with the mentioned version. If you are still experiencing this issue with our latest stable release please provide a test case and click the Reopen Issue button. Thanks!
